### ISSUE TITLE
Feature Request: Http post onProviderChange(callbackFn, failureFn)

### ISSUE THREAD

**Issue by shyamal890** [transistorsoft/cordova-background-geolocation-lt#250]
Can http post be made on onProviderChange(callbackFn, failureFn) callback?

It would great to know when a user switched off a setting even when the app is killed.

**Comment by christocracy** [comment]
Android actually does this already, recording a location and setting `event: providerchange` in the recorded location.  I'll see what can be done for iOS.

**Comment by shyamal890** [comment]
@christocracy Sorry, I didn't really understand. Do you mean to say that the change is recorded in [Location Data Schema](https://github.com/transistorsoft/cordova-background-geolocation-lt/wiki/Location-Data-Schema)? If so, how?

**Comment by christocracy** [comment]
@shyamal890 I see you commenting on Mauron85's repo.  What is your purpose?  Are you actually using my stuff or are you here to have questions answered?

**Comment by shyamal890** [comment]
The question I am asking is to solve my specific problem. I have posted the same issue on Mauron85's repo too. However, seeing he's not actively developing, I am looking for alternatives and if your solution does solve the problem mentioned above.

**Comment by christocracy** [comment]
As I said:
>Android actually does this already, recording a location and setting `event: providerchange`.   

```
---------------------------------------------------------------------
- POST /locations
 {  
  "location": {
    "activity": {
      "type": "still",
      "confidence": 100
    },
    "provider": {  // <-- location provider info
      "network": true,
      "gps": true,
      "enabled": true
    },
    "odometer": 0,
    "battery": {
      "is_charging": false,
      "level": 0.8
    },
    "event": "providerchange",  // <-- providerchange event
    "is_moving": false,
    "uuid": "cbc1b8fb-7c2d-4e1b-ab3f-5f034a4bdda1",
    "coords": {
      "latitude": 45.5192657,
      "longitude": -73.6169116,
      "accuracy": 20,
      "speed": -1,
      "heading": -1,
      "altitude": -1
    },
    "timestamp": "2017-01-17T14:53:04.222Z"
  }
}
```

iOS does **not** do this currently but *could*

**Comment by christocracy** [comment]
If I toggle Android Location Services in settings
- Device Only
- Battery Saving
- Disable location-services:

You can build the [SampleApp](https://github.com/transistorsoft/cordova-background-geolocation-SampleApp) for Android and observe this yourself

```
---------------------------------------------------------------------
- POST /locations
 {
  "location": {
    "activity": {
      "type": "still",
      "confidence": 100
    },
    "provider": {
      "network": true,
      "gps": false,
      "enabled": true
    },
    "odometer": 0,
    "battery": {
      "is_charging": false,
      "level": 0.8
    },
    "event": "providerchange",
    "is_moving": false,
    "uuid": "62c4e869-0091-40b8-b8ef-61ab4b1e1028",
    "coords": {
      "latitude": 45.5192657,
      "longitude": -73.6169116,
      "accuracy": 20,
      "speed": -1,
      "heading": -1,
      "altitude": -1
    },
    "timestamp": "2017-01-17T14:59:06.252Z"
  }
}

---------------------------------------------------------------------
- POST /locations
 {
  "location": {
    "activity": {
      "type": "still",
      "confidence": 100
    },
    "provider": {
      "network": false,
      "gps": false,
      "enabled": false
    },
    "odometer": 0,
    "battery": {
      "is_charging": false,
      "level": 0.8
    },
    "event": "providerchange",
    "is_moving": false,
    "uuid": "62c4e869-0091-40b8-b8ef-61ab4b1e1028",
    "coords": {
      "latitude": 45.5192657,
      "longitude": -73.6169116,
      "accuracy": 20,
      "speed": -1,
      "heading": -1,
      "altitude": -1
    },
    "timestamp": "2017-01-17T14:59:06.252Z"
  }
}
```

**Comment by 23tux** [comment]
Could it be, that the `provider` information is merged into the JSON post request on android? I provided a `locationTemplate` and the `provider` info got merged into it. I could only reproduce this on Android and not iOS.

(btw: if the provider info is available in the http posts and the locationTemplates, they don't show up in the docs https://github.com/transistorsoft/cordova-background-geolocation-lt/blob/master/docs/http.md#template-tags)
